### DISCUSSION TITLE
Multiple inputs data for the MIONet model.

### DISCUSSION THREAD

**Discussion by Nhatkhang** [lululxvi/deepxde#1840] [Q&A] [answered]
Hi all, 
I am trying to create a multiple input DeepONet, i.e. MIONet model, with 3+ branch net. To build a model, the data object needs to be created first. However, as I follow the readthedocs manual and the MIONet paper, it only offers up to quadruple data, deepxde.data.quadruple, or 2 branch nets only. So let say, I have an array size (N_b, dim_1) as input for branch net1, size (N_b, dim_1) as input for branch net2, ..., and so on, so in the end, it will be end up with M branch nets with the same size (N_b, dim1). The trunk net will have the size of (N_t,dim2). My output will have the size of (N_b, N_t).  Could you please guide me how I can create a data object in this case. 
Thank you in advance.

**Comment by lululxvi** [comment]
You need to create a new class similar to MIONet, but with more branches.

**Reply by Nhatkhang** [reply]
Thank you for your response. I have a follow-up question regarding DeepONet. Can it handle a 2D input function? For instance, in my current setup, the input for the branch network has a shape of (Nf, Nt), where Nf represents the number of functions (extracted from the function space) and Nt corresponds to the number of time stamps.

Now, suppose I want to input a 2D spatial distribution function, which would have a shape of (2, Nt). In this case, the input for the branch network would take the form (Nf, 2, Nt). However, when I attempted to fit this input into the original branch network, I encountered a shape mismatch error.

Could you provide guidance on how to address this issue?

**Reply by lululxvi** [reply, marked as answer]
You can find some examples at https://github.com/lu-group/

**Comment by cayobro** [comment]
@Nhatkhang I am currently facing the same issue
My input data has the shape (ns, nt, 3); or (ns, 3, nt); which corresponds to ns samples of dimension 3 x nt.
How did you end up solving this problem?
